### ISSUE TITLE
Will this work be merged to upstream plonky2 repo?

### ISSUE THREAD

**Issue by vivekvpandya** [sideprotocol/plonky2-gpu#2]
Thanks for open sourcing this work! Are there any future plans to merge upstream repo?

**Comment by liangping** [comment]
We are happy to do that, but I don't think the upstream repo will accept.

**Comment by vivekvpandya** [comment]
Why not ? I think code needs little bit of cleanup and perhaps squash commits which don't have important commit message. Also instead of my_prove() it would be better to use prove_with_cuda() or some other better naming convention.

**Comment by matthiasgoergens** [comment]
> We are happy to do that, but I don't think the upstream repo will accept.

I work with @vivekvpandya.  We can try to help you work with upstream to get your code in, if you are interested.

Btw, would you be willing to change the license to the one upstream uses?

Why do you think upstream wouldn't want your code?

**Comment by liangping** [comment]
Which upstream repo do you mean? 
We was trying to add a link on the Readme, https://github.com/0xPolygonZero/plonky2/pull/1441#event-11368077590,  but it's been rejected.

**Comment by vivekvpandya** [comment]
> Which upstream repo do you mean? We was trying to add a link on the Readme, [0xPolygonZero/plonky2#1441 (comment)](https://github.com/0xPolygonZero/plonky2/pull/1441#event-11368077590), but it's been rejected.

The same where your PR was created.
But I am curious why you want to add a link to README? why not just merge commits from this repo (after polishing) to plonky2 upstream repo?

**Comment by matthiasgoergens** [comment]
Yes, we had in mind helping you to contribute your code directly to the main plonky2 repository. Not to add a link in a readme somewhere.

We've contributed to plonk2 before, and found them fairly welcoming.

**Comment by liangping** [comment]
>But I am curious why you want to add a link to README? why not just merge commits from this repo (after polishing) to plonky2 upstream repo?

We can polish code for the merge.  This project is actually supported by Side Labs.  Do you mind if we add "Side Labs" somewhere for its early supporting.

**Comment by matthiasgoergens** [comment]
I don't mind, but we'd need to check with plonky2 folks what they think.

**Comment by mertwole** [comment]
@liangping are you guys still planning merging it upstream?

**Comment by matthiasgoergens** [comment]
Yes, but haven't worked on it much recently. We had some people join and leave.
